msal-go/index.md

```diff
@@ -0,0 +1,106 @@
+
+The latest code resides in the `dev` branch in the [library GitHub repository](https://github.com/AzureAD/microsoft-authentication-library-for-go).
+
+## Installation
+
+### Setting up Go
+
+To install Go, visit [this link](https://golang.org/dl/).
+
+### Installing MSAL Go
+
+```bash
+go get -u github.com/AzureAD/microsoft-authentication-library-for-go/
+```
+
+## Usage
+
+Before using MSAL Go, you will need to [register your application with the Microsoft identity platform](/azure/active-directory/develop/quickstart-v2-register-an-app).
+
+### Public surface
+
+The Public API of the library can be found in the following directories under `apps`.
+
+- `confidential` - The confidential application API
+- `public` - The public application API
+- `cache` - The cache interface that can be implemented to provide persistence cache storage of credentials
+
+Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+
+1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
+
+   - Initializing a public client:
+
+    ```go
+    publicClientApp, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+   - Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
+    if err != nil {
+        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+    }
+    confidentialClientApp, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+1. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
+
+    ```go
+    var userAccount public.Account
+    accounts := publicClientApp.Accounts()
+    if len(accounts) > 0 {
+        // Assuming the user wanted the first account
+        userAccount = accounts[0]
+        // found a cached account, now see if an applicable token has been cached
+        result, err := publicClientApp.AcquireTokenSilent(context.Background(), []string{"your_scope"}, public.WithSilentAccount(userAccount))
+        accessToken := result.AccessToken
+    }
+    ```
+
+1. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token.
+
+    ```go
+    result, err := publicClientApp.AcquireToken"ByOneofTheActualMethods"([]string{"your_scope"}, ...(other parameters depending on the function))
```

Comment:
Many of the wording from this document sound very familiar to me. :-)

Half joking aside, this particular line is not currently syntactically valid. Was that intentional? Do you consider write it like below?
```suggestion
    result, err := publicClientApp.AcquireTokenByOneofTheActualMethods([]string{"your_scope"}, ...(other parameters depending on the function))
```